frontend/src/app/services/storage-handler.service.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * @description
+ *  This file handles all the CRUD operations related to ionic storage
+ */
+import { Injectable } from '@angular/core';
+import { Storage } from '@ionic/storage';
+
+import { FormsData } from '../model/forms-data.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StorageHandlerService {
+  constructor(private storage: Storage) {}
+
+  /**
+   * Used to create and and add an item in the storage
+   * @param key Unique key for the item
+   * @param value Conatains the FormsData field objects
+   * @returns Promise of forms data object
+   */
+  addItem(key: string, value: DeepPartial<FormsData>): Promise<FormsData> {
+    return this.storage.ready().then(() => {
+      return this.storage.set(key, value).then((item) => {
+        return item;
+      });
+    });
+  }
+
+  /**
+   * Used to get an item from the storage
+   * @param key Unique key for the item
+   * @returns Promise of forms data object
+   */
+  getItem(key: string): Promise<FormsData> {
+    return this.storage.ready().then(() => {
+      return this.storage.get(key).then((item) => {
+        return item;
+      });
+    });
+  }
+
+  /**
+   * Used to get all items from the storage
+   * @returns Promise of array of forms data object
+   */
+  getAllItems(): Promise<FormsData[]> {
+    const items: Array<FormsData> = [];
```

Comment:
Maybe, just use FormsData[] instead of Array<FormsData>. I don't think there is a technical difference, but I got the feeling I see the first version much more often in other projects

Reply:
It does not make any difference. I simply liked the generic approach of declaring array in typescript. Hence followed it.